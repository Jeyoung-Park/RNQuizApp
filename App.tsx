/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import React from 'react';
import { SafeAreaView, StyleSheet, Text } from 'react-native';

const App = () => {
	return (
		// eslint-disable-next-line @typescript-eslint/no-use-before-define
		<SafeAreaView style={styles.container}>
			<Text>Quiz App</Text>
		</SafeAreaView>
	);
};

const styles = StyleSheet.create({
	container: {
		flex: 1,
	},
});

export default App;
