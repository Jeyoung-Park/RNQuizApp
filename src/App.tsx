/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import React from 'react';
import { SafeAreaView, StyleSheet, Text } from 'react-native';
import Home from './components/screen/Home';

const App = () => {
	return (
		// eslint-disable-next-line @typescript-eslint/no-use-before-define
		<SafeAreaView style={styles.container}>
			<Home />
		</SafeAreaView>
	);
};

const styles = StyleSheet.create({
	container: {
		flex: 1,
	},
});

export default App;
