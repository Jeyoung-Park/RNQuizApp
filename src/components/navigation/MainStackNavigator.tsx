import * as React from 'react';

// import {createStackNavigator} from '@react-navigation/stack';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import Home from '../screen/Home';
import Quiz from '../screen/Quiz';
import Result from '../screen/Result';

const Stack = createNativeStackNavigator();

const MainStackNavigator = () => (
	<Stack.Navigator initialRoute="Home">
		<Stack.Screen
			name="Home"
			component={Home}
			options={{
				headerShown: false,
			}}
		/>
		<Stack.Screen
			name="Quiz"
			component={Quiz}
			options={{
				headerShown: false,
				gestureEnabled:false,
			}}
		/>
		<Stack.Screen
			name="Result"
			component={Result}
			options={{
				headerShown: false,
				gestureEnabled:false,
			}}
		/>
	</Stack.Navigator>
);

export default MainStackNavigator;

