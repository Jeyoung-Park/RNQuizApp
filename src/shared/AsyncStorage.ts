import AsyncStorage from '@react-native-async-storage/async-storage';

interface StoreDataProps extends ReadDataProps {
	// key: string;
	value: any;
}

interface ReadDataProps {
	key: string;
}

export const storeData = async ({ key, value }: StoreDataProps) => {
	try {
		const jsonValue = JSON.stringify(value);
		await AsyncStorage.setItem(key, jsonValue);
	} catch (e) {
		// saving error
		console.warn('error storing data, ', e);
	}
};

export const getData = async ({ key }: ReadDataProps) => {
	try {
		const jsonValue = await AsyncStorage.getItem(key);
		return jsonValue != null ? JSON.parse(jsonValue) : null;
	} catch (e) {
		// error reading value
		console.warn('error reading data, ', e);
	}
};
