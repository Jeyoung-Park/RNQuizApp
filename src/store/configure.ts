import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import reducer from '../models';

const configure = () => createStore(reducer, applyMiddleware(thunk));

export default configure;
