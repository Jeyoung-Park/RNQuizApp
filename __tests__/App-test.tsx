/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';
import { render, fireEvent } from '@testing-library/react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import App from '../src/App';
import MainStackNavigator from '../src/components/navigation/MainStackNavigator';
import { getData } from '../src/shared/functions/AsyncStorage';
import { Provider } from 'react-redux';
import store from '../src/store';
// Note: test renderer must be required after react-native.
// beforeAll(() => {
// 	jest.mock('@react-native-community/async-storage');
// });

const KEY_WRONG_QUIZZES = 'KEY_WRONG_QUIZZES';

jest.mock('react-native-simple-toast', () => ({
	SHORT: jest.fn(),
}));

it('checks if Async Storage is used', async () => {
	// const dispatch = useDispatch();

	await getData({
		key: KEY_WRONG_QUIZZES,
	});

	expect(AsyncStorage.getItem).toBeCalledWith(KEY_WRONG_QUIZZES);
});

it('snapshot test', async () => {
	const tree = renderer.create(<App />).toJSON();
	expect(tree).toMatchSnapshot();
});

it('check page contains proper elements', async () => {
	const component = (
		<Provider store={store}>
			<NavigationContainer>
				<MainStackNavigator />
			</NavigationContainer>
		</Provider>
	);

	const { findByText } = render(component);

	const title = await findByText('React Native Quiz App');
	const description = await findByText('React Native로 제작된 퀴즈 앱입니다.');
	const buttonQuiz = await findByText('퀴즈 풀기');
	const buttonWrongQuiz = await findByText('오답 노트');

	expect(title).toBeTruthy();
	expect(description).toBeTruthy();
	expect(buttonQuiz).toBeTruthy();
	expect(buttonWrongQuiz).toBeTruthy();
});

// it('checks if Async Storage is used', async () => {
// 	await asyncOperationOnAsyncStorage();

// 	expect(AsyncStorage.getItem).toBeCalledWith('myKey');
// });
