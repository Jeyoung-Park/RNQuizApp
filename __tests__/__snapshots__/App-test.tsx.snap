// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        forwardedRef={[Function]}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="pop"
        stackPresentation="push"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          hidden={true}
          hideBackButton={false}
          hideShadow={false}
          largeTitleHideShadow={false}
          title="Home"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTSafeAreaView
            emulateUnlessSupported={true}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 32,
                  }
                }
              >
                React Native Quiz App
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 20,
                    "marginTop": 20,
                  }
                }
              >
                React Native로 제작된 퀴즈 앱입니다.
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "space-around",
                  "marginBottom": 50,
                  "width": "100%",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "gray",
                    "borderRadius": 20,
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingVertical": 20,
                    "width": "80%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 16,
                    }
                  }
                >
                  퀴즈 풀기
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "gray",
                    "borderRadius": 20,
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingVertical": 20,
                    "width": "80%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 16,
                    }
                  }
                >
                  오답 노트
                </Text>
              </View>
            </View>
          </RCTSafeAreaView>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </RNCSafeAreaProvider>
</RCTSafeAreaView>
`;
